src/soilbiogeochem/SoilBiogeochemNitrifDenitrifMod.F90

```diff
@@ -278,9 +282,20 @@ subroutine SoilBiogeochemNitrifDenitrif(bounds, num_soilc, filter_soilc, &
                else
                   om_frac = 1._r8
                end if
-               diffus (c,j) = (d_con_g(2,1) + d_con_g(2,2)*t_soisno(c,j)) * 1.e-4_r8 * &
-                    (om_frac * f_a**(10._r8/3._r8) / watsat(c,j)**2 + &
-                    (1._r8-om_frac) * eps**2 * f_a**(3._r8 / bsw(c,j)) ) 
+
+               ! Diffusitivity after Moldrup et al. (2003)
+               ! Eq. 8 in Riley et al. (2011, Biogeosciences)
+               diffus_moldrup = fc_air_frac**2 * fc_air_frac_as_frac_porosity**(3._r8 / bsw(c,j))
+
+               ! Diffusivity after Millington & Quirk (1961)
+               ! Eq. 9 in Riley et al. (2011, Biogeosciences)
+               diffus_millingtonquirk = fc_air_frac**(10._r8/3._r8) / watsat(c,j)**2
+
+               ! First, get diffusivity as a unitless constant, which is what's needed to
+               ! calculate ratio_k1 below.
+               diffus (c,j) = &
+                    (om_frac        * diffus_millingtonquirk + &
+                    (1._r8-om_frac) * diffus_moldrup ) 
```

Comment:
I like how this breaks out the two diffusing calculations following equations in the paper, but what happened to the first part of the diffus equation `(d_con_g(2,1) + d_con_g(2,2)*t_soisno(c,j)) * 1.e-4_r8 *`?

Reply:
Oh, I see it's added with calculation of `D0`, below.

Reply:
Now I'm confused because  `D0` is being calculated on the `else` side of the if `(use_lch4)` logical?  Is the intended?

Reply:
`D0` is actually being calculated at line 389, which is _after_ that if-else block (ends at line 313). The code at line 390 needs to happen after the calculation of `ratio_k1` at line 381.

In the "else" case (i.e., `use_lch4` false), `diffus` is zero by line 390 so it doesn't matter what `D0` is.